hedera-node/src/main/java/com/hedera/services/ledger/MerkleAccountScopedCheck.java

```diff
@@ -156,26 +168,16 @@ public ResponseCodeEnum validateNftAllowance(
 			@Nullable final Function<AccountProperty, Object> extantProps,
 			final Map<AccountProperty, Object> changeSet) {
 		if (balanceChange.isApprovedAllowance()) {
-			final var nftAllowances = (Map<FcTokenAllowanceId, FcTokenAllowance>) getEffective(
-					NFT_ALLOWANCES, account, extantProps, changeSet);
-			final var nftAllowance = nftAllowances.getOrDefault(
-					FcTokenAllowanceId.from(
-							balanceChange.getToken().asEntityNum(), EntityNum.fromAccountId(balanceChange.getPayerID())),
-					null
-			);
+			final var approveForAllNftsAllowances = (Set<FcTokenAllowanceId>) getEffective(
+					APPROVE_FOR_ALL_NFTS_ALLOWANCES, account, extantProps, changeSet);
+			final var nftAllowanceId = FcTokenAllowanceId.from(
+							balanceChange.getToken().asEntityNum(), EntityNum.fromAccountId(balanceChange.getPayerID()));
 
-			if (nftAllowance == null) {
-				return SPENDER_DOES_NOT_HAVE_ALLOWANCE;
-			} else {
-				final var allowAll = nftAllowance.isApprovedForAll();
-				final var allowedSerialNums = nftAllowance.getSerialNumbers();
-
-				if (allowAll) {
-					return OK;
-				} else {
-					if (!allowedSerialNums.contains(balanceChange.serialNo())) {
-						return SPENDER_DOES_NOT_HAVE_ALLOWANCE;
-					}
+			if (!approveForAllNftsAllowances.contains(nftAllowanceId)) {
+				final var approvedSpender = (EntityId) nftsLedger.get(balanceChange.nftId(), SPENDER);
+
+				if (!approvedSpender.toGrpcAccountId().equals(balanceChange.getPayerID())) {
```

Comment:
We can save an `AccountID` creation by just using [`EntityId.matches(AccountID)`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/state/submerkle/EntityId.java#L134) here.